elyra/pipeline/pipeline.py

```diff
@@ -260,20 +262,32 @@ def __init__(
             raise ValueError("Invalid pipeline operation: Missing field 'operation filename'.")
         if not component_props.get("runtime_image"):
             raise ValueError("Invalid pipeline operation: Missing field 'operation runtime image'.")
-        if component_props.get("cpu") and not self._validate_range(component_props.get("cpu"), min_value=1):
-            raise ValueError("Invalid pipeline operation: CPU must be a positive value or None")
+        if component_props.get("cpu_request") and not self._validate_range(
+            component_props.get("cpu_request"), min_value=1
+        ):
+            raise ValueError("Invalid pipeline operation: CPU request must be a positive value or None")
+        if component_props.get("cpu_limit") and not self._validate_range(component_props.get("cpu_limit"), min_value=1):
```

Comment:
I think the limit values should be dependent upon the existence of the corresponding request value.  That is, limits do not come into play unless the corresponding request has a value.  If we make that statement, then the `min_value` on the limits should be the request value and we have our check to ensure limits are >= requests (which I don't see in this PR).

This rule would also prevent only limits from getting set, which I think we should avoid as well.

Reply:
We have some users that leave default values for requests but want to make sure they control the maximum amount of resources that a pod uses (in case it gets stuck or is not behaving as expected), so they set a limit explicitly. I think have both optional is good, but I agree that there should be a check that the limit is equal or greater than the request. What do you think?
I can also see my tests don't work as they would only fail at the argo workflow level (which I tested upon build), so need to change them to be effective

Reply:
Okay fixed this which solves the test issue too 👍🏽